packages/rich-text/src/plugins/Table/index.tsx

```diff
@@ -199,11 +207,66 @@ function hasHeadersOutsideFirstRow(nodes: CustomElement[]) {
     .some(({ children }) => (children as CustomElement[]).some(isTableHeaderCell));
 }
 
+const isAtEndOfTextSelection = (editor: SPEditor) =>
+  editor.selection?.focus.offset === getText(editor, editor.selection?.focus.path).length;
+
+function currentSelectionStartsTableCell(editor: SPEditor): boolean {
+  const [tableCellNode, path] = getNodeEntryFromSelection(editor, [
+    BLOCKS.TABLE_CELL,
+    BLOCKS.TABLE_HEADER_CELL,
+  ]);
+  return (tableCellNode && !getText(editor, path)) || editor.selection?.focus.offset === 0;
+}
+
+/**
+ * This traversal strategy is unfortunately necessary because Slate doesn't
+ * expose something like Node.next(editor).
```

Comment:
:scream: there's no helpers for that provided anywhere?

Reply:
`Path.next` "does not ensure that the returned `Path` is valid in the document" [[1](https://docs.slatejs.org/api/locations/path#path.next-path-path-greater-than-path)]. In practice this means it's programmatically +1s the last number in the path given and is really only useful for placing new nodes ahead of the current one within a path.